error_test.go

```diff
@@ -326,178 +334,226 @@ var _ = Describe("Merge", func() {
+								Ω(mErr.(*ErrorResponse).Meta[k]).Should(Equal(v))
+							}
+						})
 					})
 				})
 
-				Context("with other metadata", func() {
-					metaValues2 := map[string]interface{}{"foo": 1, "bar": 2}
+				Context("with metadata with a common key", func() {
+					const commonKey = "foo"
+
+					var metaValues = map[string]interface{}{commonKey: "bar", "foo2": 44}
+					var metaValues2 = map[string]interface{}{commonKey: 43, "baz": 42}
 
 					BeforeEach(func() {
+						mv := make(map[string]interface{}, len(metaValues))
+						for k, v := range metaValues {
+							mv[k] = v
+						}
+						err.(*ErrorResponse).Meta = mv
 						mErr2.Meta = metaValues2
 					})
 
 					It("merges the metadata", func() {
-						Ω(mErr.Meta).Should(HaveLen(len(metaValues) + len(metaValues2)))
+						Ω(mErr.(*ErrorResponse).Meta).Should(HaveLen(len(metaValues) + len(metaValues2) - 1))
 						for k, v := range metaValues {
-							Ω(mErr.Meta[k]).Should(Equal(v))
+							if k != commonKey {
+								Ω(mErr.(*ErrorResponse).Meta[k]).Should(Equal(v))
+							}
 						}
 						for k, v := range metaValues2 {
-							Ω(mErr.Meta[k]).Should(Equal(v))
+							if k != commonKey {
+								Ω(mErr.(*ErrorResponse).Meta[k]).Should(Equal(v))
+							}
 						}
+						Ω(mErr.(*ErrorResponse).Meta[commonKey]).Should(Equal(metaValues2[commonKey]))
 					})
 				})
 			})
+		})
 
-			Context("with metadata with a common key", func() {
-				const commonKey = "foo"
+		Context("with MergeableError", func() {
+			BeforeEach(func() {
+				err = &MergeableErrorResponse{ErrorResponse: &ErrorResponse{Detail: detail, Status: status, Code: code}}
+			})
 
-				var metaValues = map[string]interface{}{commonKey: "bar", "foo2": 44}
-				var metaValues2 = map[string]interface{}{commonKey: 43, "baz": 42}
+			Context("with a nil argument", func() {
+				It("returns the target", func() {
+					Ω(mErr).Should(Equal(err))
+				})
+			})
+
+			Context("with Error", func() {
+				const detail2 = "foo2"
+				var status2 = status
+				var code2 = code
+				var mErr2 *ErrorResponse
 
 				BeforeEach(func() {
-					mv := make(map[string]interface{}, len(metaValues))
-					for k, v := range metaValues {
-						mv[k] = v
-					}
-					err.(*ErrorResponse).Meta = mv
-					mErr2.Meta = metaValues2
+					mErr2 = &ErrorResponse{Detail: detail2, Status: status2, Code: code2}
+					err2 = mErr2
 				})
 
-				It("merges the metadata", func() {
-					Ω(mErr.Meta).Should(HaveLen(len(metaValues) + len(metaValues2) - 1))
-					for k, v := range metaValues {
-						if k != commonKey {
-							Ω(mErr.Meta[k]).Should(Equal(v))
-						}
-					}
-					for k, v := range metaValues2 {
-						if k != commonKey {
-							Ω(mErr.Meta[k]).Should(Equal(v))
-						}
-					}
-					Ω(mErr.Meta[commonKey]).Should(Equal(metaValues2[commonKey]))
+				It("calls user defined merge", func() {
+					Ω(mErr.(*MergeableErrorResponse).MergeCalled).Should(Equal(1))
+				})
+			})
+
+			Context("with MergableError", func() {
```

Comment:
would be good to also have a test where the first error is NOT a mergeable error but the second one is that validates the second error `Merge` method is called.